internal/apply/apply.go

```diff
@@ -89,10 +110,58 @@ func applyAll(dir string, projectConfig projectconfig.ZeroProjectConfig, applyEn
 		envVarTranslationMap := modConfig.GetParamEnvVarTranslationMap()
 		envList = util.AppendProjectEnvToCmdEnv(mod.Parameters, envList, envVarTranslationMap)
 		flog.Debugf("Env injected: %#v", envList)
-		flog.Infof("Executing apply command for %s...", modConfig.Name)
-		util.ExecuteCommand(exec.Command("make"), modulePath, envList)
+
+		// only print msg for apply, or else it gets a little spammy
+		if lifecycleName == "apply" {
+			flog.Infof("Executing %s command for %s...", lifecycleName, modConfig.Name)
+		}
+		operationCommand := getModuleOperationCommand(modConfig, operation)
+		execErr := util.ExecuteCommand(exec.Command(operationCommand[0], operationCommand[1:]...), modulePath, envList)
+		if execErr != nil {
+			formatedErr := errors.New(fmt.Sprintf("Module (%s) %s", modConfig.Name, execErr.Error()))
+			if bailOnError {
+				return formatedErr
+			} else {
+				moduleErrors = append(moduleErrors, formatedErr)
+			}
+		}
 		return nil
 	})
+	if err != nil {
+		moduleErrors = append(moduleErrors, err)
+	}
+
+	return moduleErrors
+}
+
+func getModuleOperationCommand(mod moduleconfig.ModuleConfig, operation string) (operationCommand []string) {
+	defaultCheck := []string{"make", "check"}
+	defaultApply := []string{"make"}
+	defaultSummary := []string{"make", "summary"}
+
+	switch operation {
+	case "check":
+		if mod.Commands.Check != "" {
+			operationCommand = strings.Split(mod.Commands.Check, " ")
```

Comment:
You can use `strings.Fields()` for this. The only thing is it doesn't handle stuff like quotes, which we would need another library for, but it should be good enough. 
https://golang.org/pkg/strings/#Fields

Reply:
is there an easy to to handle `|`s and `>` s  if users put some advance statement in the command?

Reply:
Let's just go with Fields and encourage people to put any logic into a script